### PR TITLE
Create one avalon sdk for all clients(worker service, connector and requesters)

### PR DESCRIPTION
**Pull request by Ram-srini**
…ester)

1. Combine connector and client sdk to avalon sdk
2. Update all the application with the new sdk
3. Organize the utility modules inside the SDK.

Signed-off-by: Ramakrishna Srinivasamurthy <ramakrishna.srinivasamurthy@intel.com>

### CODE REVIEW

File: sdk/avalon_sdk/direct/jrpc/jrpc_util.py [hunk 1]
```diff
@@ -48,11 +26,12 @@ class JsonRpcErrorCode(Enum):
     UNSUPPORTED_MODE = 6
     KEY_EXISTS = 9
 
+def construct_message(status, message):
```
Comment (manojgop) [inline sdk/avalon_sdk/direct/jrpc/jrpc_util.py]: This method name looks very generic for the functionality it implements. May be rename it to create_json_status_message ?

File: sdk/avalon_sdk/ethereum/ethereum_worker_registry.py [hunk 2]
```diff
@@ -40,6 +41,112 @@ def __init__(self, config):
...
+        tcf_home = environ.get("TCF_HOME", "../../../")
+        contract_file_name = tcf_home + "/" + \
+            config["ethereum"]["worker_registry_contract_file"]
+        contract_address = \
+            config["ethereum"]["worker_registry_contract_address"]
+        self.__contract_instance = self.__eth_client.get_contract_instance(
+            contract_file_name, contract_address
+        )
+
     def worker_register(
```
Comment (manojgop) [inline sdk/avalon_sdk/ethereum/ethereum_worker_registry.py]: Can we organize the functions in file such that all public functions are followed by private functions or vice-versa. But let's follow a consistent ordering.

Comment (Ram-srini) [reply sdk/avalon_sdk/ethereum/ethereum_worker_registry.py]: Does it really required?

File: sdk/avalon_sdk/direct/jrpc/jrpc_util.py (hunk 1 above)
Comment (Ram-srini) [reply sdk/avalon_sdk/direct/jrpc/jrpc_util.py]: Done

File: sdk/avalon_sdk/ethereum/ethereum_worker_registry.py [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+                is_valid = validate_details(details)
+                if is_valid is not None:
+                    return construct_json_resp_message("failed", is_valid)
+
+            txn_hash = self.__contract_instance.functions.workerRegister(
+                worker_id, worker_type.value, org_id, application_ids,
+                details).buildTransaction(
+                    self.__eth_client.get_transaction_params()
+                )
+            tx = self.__eth_client.execute_transaction(txn_hash)
```
Comment (manojgop) [inline sdk/avalon_sdk/ethereum/ethereum_worker_registry.py]: execute_transaction() function returns json status response. So better to name variable "tx" something like  "tx_status" to make the variable name look more meaningful

File: sdk/avalon_sdk/ethereum/ethereum_worker_registry.py [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+        contract_address = \
+            config["ethereum"]["worker_registry_contract_address"]
+        self.__contract_instance = self.__eth_client.get_contract_instance(
+            contract_file_name, contract_address
+        )
+
+    def worker_register(
+            self, worker_id, worker_type, org_id, application_ids, details):
+        """
+        Register new worker with details of worker
```
Comment (manojgop) [inline sdk/avalon_sdk/ethereum/ethereum_worker_registry.py]: Also mention the return value of the function. This function return status response formatted as JSON string.

File: sdk/avalon_sdk/interfaces/work_order_proxy.py [hunk 5]
```diff
@@ -13,42 +13,44 @@
...
-        provided, requesterId below is used as a key.
-        4.signature is a signature generated by
-        the worker on the worker_id, tag
-        nonce.
+        This function is called by the Worker Service to
+        complete a Work Order successfully or in error.
+        This API is for proxy model.
+        params
+            ork_order_id is unique id to identify the work order request
+            ork_order_response is the Work Order response data in string
```
Comment (manojgop) [inline sdk/avalon_sdk/interfaces/work_order_proxy.py]: typo. ork_order_id -> work_order_id
ork_order_response -> work_order_response

File: sdk/avalon_sdk/ethereum/ethereum_worker_registry.py (hunk 4 above)
Comment (Ram-srini) [reply sdk/avalon_sdk/ethereum/ethereum_worker_registry.py]: Changed to return None or error and on success return txn receipt.

File: sdk/avalon_sdk/ethereum/ethereum_worker_registry.py (hunk 3 above)
Comment (Ram-srini) [reply sdk/avalon_sdk/ethereum/ethereum_worker_registry.py]: Done

File: sdk/avalon_sdk/interfaces/work_order_proxy.py (hunk 5 above)
Comment (Ram-srini) [reply sdk/avalon_sdk/interfaces/work_order_proxy.py]: Done

**Comment by Ram-srini** [comment]
> lint_module sdk "$SINCE" 1 0 0 0 || rv=1

@danintel I have added  lint check for my new module and fixed the lint errors. Please review and approve.

**Comment by Ram-srini** [comment]
@danintel Can you please approve this PR? This SDK PR required for proxy model implementation.

**Comment by danintel** [comment]
Approved.

**Pull request merged**